### PR TITLE
Improve error handling in Tools load

### PR DESCRIPTION
**Pull request by Slayer95**

### CODE REVIEW

**Comment by ascriptmaster** [comment]
Speak performance to me. How much faster does this make the code? Obviously it makes it faster because functions won't be deoptimized, but by how much?

**Comment by Slayer95** [comment]
This has no significative performance benefit by itself. However, it's a required change for getting the best from further optimizations.

E.g. Switching the ``forEach`` calls to ``for`` loops would imply a benefit of up to 34% speed on the tip of #2033, compared to 10% on ``master``.

**Comment by Zarel** [comment]
Things that are only run a constant number of times really don't need to be optimized, do they?

**Comment by Slayer95** [comment]
If they lock down the server for more than a second while it's supposed to be performing I/O, then I think it should (hotpatch).

Also, it wouldn't hurt to cut down server loading time.

**Comment by Slayer95** [comment]
y/n

File: tools.js [hunk 1]
```diff
@@ -59,6 +59,45 @@ module.exports = (function () {
 		timid: {name:"Timid", plus:'spe', minus:'atk'}
 	};
 
+	function tryRequire (path) {
```
Comment (Zarel) [inline tools.js]: In theory our convention is `tryRequire(`.

We require a space after the `function` keyword, but no space between the function name and `(`.

**Comment by Slayer95** [comment]
Updated

(TODO: stricter ``jscs`` settings for named functions)

File: tools.js [hunk 2]
```diff
@@ -130,20 +173,19 @@ module.exports = (function () {
 			});
 		}
 	}
+
 	Tools.loadMods = function () {
```
Comment (Zarel) [inline tools.js]: This function appears to be a ton of try blocks in a row. I think it's consolidated enough not to need the functions split out...

Comment (Slayer95) [inline tools.js]: Hmm, I guess...
I intend to replace this function by this other though https://github.com/Slayer95/Pokemon-Showdown/blob/tools-split/tools.js#L913-L979 (which allows to opt-out of Tools data load by removing the ``info.js`` plugin, and to use Tools as... tools)

Inclusion of this change here is mainly to allow a subsequent cleaner diff

Comment (Slayer95) [inline tools.js]: On second thought, this might be contraproducent.
Ok, will revert this part. I have to rebase anyway...

**Comment by Zarel** [comment]
`tryRequire` doesn't seem to be done very well. Why not just return `null` if the require fails?

**Comment by Slayer95** [comment]
For two reasons:
- An existent file required can export ``null``.
- Just returning ``null`` loses information on the exception thrown, and prevents its proper handling. In particular, it prevents discriminating between errors with code ``MODULE_NOT_FOUND`` and errors that are ``SyntaxError``.

**Comment by Zarel** [comment]
There is no reason for a `data/` file to export `null`, and if it does, I don't think we'd treat it differently from a nonexistent file, anyway.

We could return `null` and `undefined` or something to discriminate between those errors.

In general I just don't like the idea of allocating a new object for something like this.

**Comment by Slayer95** [comment]
Ok, I guess we can return the error...

**Comment by Slayer95** [comment]
I don't think this stood by itself anymore, so I went one step further.

**Pull request merged**